File: android/app/src/main/java/com/yfremote/android/server/models/FileUploadResponse.kt
package com.yfremote.android.server.models

import kotlinx.serialization.Serializable

// Spiegelt Models/FileUploadResponse.cs.
@Serializable
data class FileUploadResponse(
    val success: Boolean,
    val fileName: String? = null,
    val error: String? = null,
) {
    companion object {
        fun ok(fileName: String) = FileUploadResponse(success = true, fileName = fileName)

        fun fail(error: String) = FileUploadResponse(success = false, error = error)
    }
}

File: Models/FileUploadResponse.cs
namespace YFRemote.Server.Models;

public sealed record FileUploadResponse(bool Success, string? FileName = null, string? Error = null)
{
    public static FileUploadResponse Ok(string fileName) => new(true, FileName: fileName);

    public static FileUploadResponse Fail(string error) => new(false, Error: error);
}
